addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java

```diff
@@ -759,7 +759,14 @@ public void run() {
                 }
 
                 pong = false;
-                getRpcClient(getDefaultInterface()).ping(getDefaultInterface(), id);
+                if (config.getGatewayInfo().isCCU1()) {
+                    // the CCU1 does not support the ping command, we need a workaround
+                    getRpcClient(getDefaultInterface()).listBidcosInterfaces(getDefaultInterface());
+                    // if there is no exception, connection is valid and we have to simulate a received pong
+                    pongReceived();
```

Comment:
Given its not actually a pong which was received it would make me way more happy to restructure it like follows:

```java
void pongReceived() {
    connectionConfirmed();
}

private void connectionConfirmed() {
    pong = true;
    if (connectionLost) {
        connectionLost = false;
        logger.info("Connection resumed on gateway '{}'", id);
        gatewayAdapter.onConnectionResumed();
    }
}
```

What do you think?